File: 2015/11 Corporate Policy/solution.c
#include <stdio.h>

int main(int argc, char *argv[]) {
    char value[] = "hxcaabcc"; //"hxbxwxba";
    char *v, *e = value + 8;
    for(;;) {
        for(v = e; (v != value) && ++*(--v) == 'z'; ) *v = 'a';
        for(v = value; (v != value + 6) && ((v[0] != v[1] - 1) || (v[0] != v[2] - 2)); v++);
        if (v == value + 6) continue;
        for(v = value; (v != value + 5) && (v[0] != v[1]); v++);
        if (v == value + 5) continue;
        for(v+=2; (v != value + 7) && (v[0] != v[1]); v++);
        if (v == value + 7) continue;
        printf("%s\n", value);
        break;
    }
}

File: 2015/11 Corporate Policy/solution.py
import re

doubles = re.compile("(.)\\1.*(.)\\2")
illegal = re.compile("[ilo]")
value = list("hxbxwxba")
value = list("hxcaabcc")

while True:
    for i in range(7, -1, -1):
        if value[i] == 'z':
            value[i] = 'a'
        else:
            value[i] = chr(ord(value[i]) + 1)
            break
    vals = "".join(value)
    if doubles.search(vals) and not illegal.search(vals) and \
            any(ord(a) == ord(b) - 1 == ord(c) - 2
                for a, b, c in zip(value, value[1:], value[2:])):
        break
print("".join(value))
